src/main/java/fi/vrk/xroad/monitor/actor/ResultCollectorActor.java

```diff
@@ -0,0 +1,110 @@
+package fi.vrk.xroad.monitor.actor;
+
+import akka.actor.AbstractActor;
+import fi.vrk.xroad.monitor.parser.SecurityServerInfo;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.ToString;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.context.annotation.Scope;
+import org.springframework.stereotype.Component;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Actor for collecting security server monitoring data processing results
+ */
+@Component
+@Scope("prototype")
+@Slf4j
+public class ResultCollectorActor extends AbstractActor {
+
+  private Map<SecurityServerInfo, MonitorDataResult> trackResults;
```

Comment:
A bit simpler "Set<SecurityServerInfo> awaitedResults" would probably have been enough for the tracking requirements (not that there's anything wrong with this one either).